src/main/java/com/palantir/gradle/versions/VersionsPropsPlugin.java

```diff
@@ -147,21 +147,21 @@ private static void setupConfiguration(
             }
 
             conf.extendsFrom(rootConfiguration);
-        });
-
-        // We must allow unifiedClasspath to be resolved at configuration-time.
-        if (VersionsLockPlugin.UNIFIED_CLASSPATH_CONFIGURATION_NAME.equals(conf.getName())) {
-            return;
-        }
 
-        // Add fail-safe error reporting
-        conf.getIncoming().beforeResolve(resolvableDependencies -> {
-            if (GradleWorkarounds.isConfiguring(subproject.getState())) {
-                throw new GradleException(String.format("Not allowed to resolve %s at "
-                        + "configuration time (https://guides.gradle.org/performance/"
-                        + "#don_t_resolve_dependencies_at_configuration_time). Please upgrade your "
-                        + "plugins and double-check your gradle scripts (see stacktrace)", conf));
+            // We must allow unifiedClasspath to be resolved at configuration-time.
+            if (VersionsLockPlugin.UNIFIED_CLASSPATH_CONFIGURATION_NAME.equals(conf.getName())) {
+                return;
             }
+
+            // Add fail-safe error reporting
+            conf.getIncoming().beforeResolve(resolvableDependencies -> {
+                if (GradleWorkarounds.isConfiguring(subproject.getState())) {
+                    throw new GradleException(String.format("Not allowed to resolve %s at "
+                            + "configuration time (https://guides.gradle.org/performance/"
+                            + "#don_t_resolve_dependencies_at_configuration_time). Please upgrade your "
+                            + "plugins and double-check your gradle scripts (see stacktrace)", conf));
+                }
+            });
```

Comment:
This move is safe.
This whole parent block (`withDependencies`) is guaranteed to be executed (by `runDependencyActions` in the linked code below) before the `beforeResolve` action is run.

https://github.com/gradle/gradle/blob/ed58dbc5366ddc542a655810235f68e83cf82584/subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/configurations/DefaultConfiguration.java#L620-L624